internal/locate/region_request.go

```diff
@@ -1087,13 +1087,25 @@ func (s *replicaSelector) onSendFailure(bo *retry.Backoffer, err error) {
 	s.state.onSendFailure(bo, s, err)
 }
 
-func (s *replicaSelector) onDeadlineExceeded() {
-	if target := s.targetReplica(); target != nil {
-		target.deadlineErrUsingConfTimeout = true
+func (s *replicaSelector) onReadReqConfigurableTimeout(req *tikvrpc.Request) bool {
+	if req.MaxExecutionDurationMs >= uint64(client.ReadTimeoutShort.Milliseconds()) {
```

Comment:
It seems we don't check like if `rea.stmtType == Read`, if the session variable of `kv timeout` is set to a small value like `1s`, the read requests from write statements would still take effect? For example, the batch get requests coming from the `INSERT` statements.

Reply:
Nice catch. But currently, the write statement doesn't set `kv_read_timeout` in TiDB.

Or just in case, we should add `stmtType` field and check `req.stmtType == Read`?

Reply:
But check `stmtType` in `client-go` is a little bit strange. If the write statement should not use `kv_read_timeout`, then the write statement should not set it.

Reply:
Could the `request source` be used to check the statement source? For example, check if it's a `select` source type.
It's more robust to do all the necessary checks to avoid possible mis-usages though users may have some assumptions on the application side.

Reply:
Check if it's `select` is not work well. Such as `explain analyze select ...`, the `request source` will be `ExplainAnalyzeSQL`.